modules/signals/rxjs-interop/src/rx-method.ts

```diff
@@ -24,22 +25,32 @@ export function rxMethod<Input>(
   }
 
   const injector = config?.injector ?? inject(Injector);
-  const destroyRef = injector.get(DestroyRef);
   const source$ = new Subject<Input>();
-
   const sourceSub = generator(source$).subscribe();
-  destroyRef.onDestroy(() => sourceSub.unsubscribe());
 
-  const rxMethodFn = (input: Input | Signal<Input> | Observable<Input>) => {
+  const rxMethodFn = (
+    input: Input | Signal<Input> | Observable<Input>,
+    customInjector?: Injector
+  ) => {
     if (isSignal(input)) {
+      const callerInjector = getCallerInjectorIfAvailable();
+      const instanceInjector = customInjector ?? callerInjector ?? injector;
+
       const watcher = effect(
         () => {
           const value = input();
           untracked(() => source$.next(value));
         },
-        { injector }
+        { injector: instanceInjector }
       );
-      const instanceSub = { unsubscribe: () => watcher.destroy() };
+
+      instanceInjector.get(DestroyRef).onDestroy(() => {
+        sourceSub.unsubscribe();
+      });
```

Comment:
when we unsubscribe from the root/source subscription, rxMethod won't be alive anymore

with this logic, unsubscribing from the source subscription will happen when the first instance injector is destroyed

Reply:
Done